doc: readme: fix example conversion configs: all -> all_topics

ottojo opened the pull request:
this was changed in #1480, see issue #1660

ottojo committed:
fix example conversion configs: all -> all_topics

Signed-off-by: Jonas Otto <jonas@jonasotto.com>

fujitatomoya reviewed (approved):
good eye! lgtm.

@MichaelOrlov @Barry-Xu-2018 this actually breaks the user configuration as reported. i will go through the documentation again.

Barry-Xu-2018 commented:
The `RecordOptions` has already been changed quite a bit.  
Refer to the code below.

https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/rosbag2_transport/src/rosbag2_transport/record_options.cpp#L58-L103

The current README.md only mentions part of it. Is it necessary to point to the relevant code, so users know what can be configured ? @fujitatomoya @MichaelOrlov

https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/README.md?plain=1#L221-L240

Barry-Xu-2018 commented:
@ottojo 

Additionally.  ` exclude` should be changed to `exclude_regex`.  
Please refer to https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/rosbag2_transport/src/rosbag2_transport/record_options.cpp#L58-L103

ottojo committed:
update topic and service selection arguments in conversion config in readme

Signed-off-by: Jonas Otto <jonas@jonasotto.com>

ottojo commented:
@Barry-Xu-2018 thanks for pointing me to the corresponding parts in the source. I updated the `exclude` key and added the other keys relevant for topic and service selection. I did not add all keys found in the source, i think some may already be excluded on purpose, because they are not relevant for bag conversion (such as for example `disable_keyboard_controls`, `compression_threads_priority` and `topic_polling_interval`)

ottojo commented:
oh, i also just notice that it's not only the `RecordingOptions` but also `StorageOptions`, where the actual options are that i wanted to use prior to hitting this issue:
https://github.com/ros2/rosbag2/blob/5a06430cac7e823e5b5522a67ce7bd3b325e516f/rosbag2_storage/src/rosbag2_storage/storage_options.cpp#L44-L56

fujitatomoya reviewed (approved):
lgtm.

i do not think we need to introduce all options here, but all options need to be aligned with the implementation.

MichaelOrlov commented on README.md:
```diff
@@ -247,7 +253,7 @@ $ ros2 bag convert -i bag1 -i bag2 -o out.yaml
 output_bags:
 - uri: merged_bag
-  all: true
+  all_topics: true
```
```suggestion
  all_topics: true
  all_services: true
```

MichaelOrlov commented on README.md:
```diff
@@ -271,7 +277,7 @@ $ ros2 bag convert -i bag1 -o out.yaml
 output_bags:
 - uri: compressed
-  all: true
+  all_topics: true
```
```suggestion
  all_topics: true
  all_services: true
```

MichaelOrlov reviewed (commented):
I think it will be valuable to add `all_services: true` for examples with bag compression and bag split. 
To avoid other potential questions in the future, like "Why services were lost during conversion?"

ottojo committed:
add all_services: true to first example

Co-authored-by: Michael Orlov <morlovmr@gmail.com>
Signed-off-by: Jonas Otto <jonas@jonasotto.com>

ottojo committed:
add all_services: true to second example

Co-authored-by: Michael Orlov <morlovmr@gmail.com>
Signed-off-by: Jonas Otto <jonas@jonasotto.com>

MichaelOrlov reviewed (approved):
LGTM. Thanks for the PR!

clalancette reviewed (approved):
Looks good to me as well.

Given that this is a change to documentation, we can merge this without CI, but @MichaelOrlov I'll wait for you to do that.  I think we should also backport this to the `jazzy` branch.

MichaelOrlov commented:
Merging without CI run since changes only in the `readme.md` file.

MichaelOrlov commented:
https://github.com/Mergifyio backport jazzy
